04_numeronym/nickolee/numeronym.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	result := make([]string, len(vals))
+	for i, val := range vals {
+		result[i] = numeronym(val) // build results slice
+	}
+	return result
+}
+
+func numeronym(s string) string {
+	// convert string to slice of runes (Unicode int32) to avoid some kind of "protocol error"
+	runes := []rune(s)
```

Comment:
I am not sure what's happening here, the expected input is string, not runes. Perhaps I am missing something here.

Reply:
Without doing this - the code will not work as expected for Unicode characters for example emojis 😅😌😡😶🤤